packages/v3-sdk/src/functions/hasInvolvedToken.ts

```diff
@@ -0,0 +1,6 @@
+import { Token } from '@pancakeswap/sdk'
+import { PoolState } from './getPool'
+
+export const hasInvolvedToken = (pool: PoolState, token: Token) => {
```

Comment:
let's use duck typing for more flexibility
```suggestion
export const involvesToken = (pool: Pick<PoolState, 'token0' | 'token1'>, token: Token) => {
```